### DISCUSSION TITLE
Brick Nextion Display and Fix - Don't unplug while updating Nextion firmware

### DISCUSSION THREAD

**Discussion by ielbury** [HASwitchPlate/HASPone#14] [General]
I unplugged my Nextion while upgrading the Nextion firmware file and after, it would boot to a black screen and serial communications would not work again with HASPOne or the Nextion editor.

To resolve, I formatted a SD card  (FAT32), copied the HASwitchPlate.tft file to the card, inserted into the display and powered up and the HASwitchPlate.tft file flashed Ok and serial communications works again.

This might be good to add to the new documentation?

**Comment by IIAIronWolf** [comment]
> I unplugged my Nextion while upgrading the Nextion firmware file and after, it would boot to a black screen and serial communications would not work again with HASPOne or the Nextion editor.
> 
> To resolve, I formatted a SD card (FAT32), copied the HASwitchPlate.tft file to the card, inserted into the display and powered up and the HASwitchPlate.tft file flashed Ok and serial communications works again.
> 
> This might be good to add to the new documentation?

I believe not powering off the device during an LCD update has been part of the documentation for quite sometime, and definitely before HASPone. Just don't power down things that are updating? haha

**Reply by ielbury** [reply]
I am not sure where in the documentation it states the repercussions and how to resolve the issue if you do lose power while updating for some reason.

https://github.com/aderusha/HASwitchPlate/blob/master/Documentation/02_Nextion_HMI.md .

I am just trying to provide the community with a solution for something I could not find with a quick Google or in the HASP documentation. If this is not important to you and this is so apparent then what's the point of your comment.

**Reply by maromaro70** [reply]
Thank you, I tried this on a crazy-acting unit, didn't work but it likely says that it's not simply a firmware issue, I think.

**Reply by IIAIronWolf** [reply]
> I am not sure where in the documentation it states the repercussions and how to resolve the issue if you do lose power while updating for some reason.
> 
> https://github.com/aderusha/HASwitchPlate/blob/master/Documentation/02_Nextion_HMI.md .
> 
> I am just trying to provide the community with a solution for something I could not find with a quick Google or in the HASP documentation. If this is not important to you and this is so apparent then what's the point of your comment.

Whoa. Calm down bud. I understand what you’re doing and that’s fine. It tells you in the old docs that if you lose power during an LCD firmware upgrade that you have to pull it out of the wall and insert an SD card with the firmware.

**Reply by ielbury** [reply]
Where? I gave the link to HMI documentation? As far as I am concerned, your statement is unsubstantiated.

**Reply by aderusha** [reply]
Allright, settle down everyone :). Not power cycling a device during firmware operations is always best practice, but as a developer one can’t always plan on your plans going to plan, so we also need a plan B for when things go sideways.  The approach presented in the OP is, in my experience, pretty bulletproof (if a little annoying due to having to pull the thing apart).

I really enjoy the enthusiasm here, let’s keep in mind we’re all working on the same goal - havin fun screwing around with cool home automation tools!

**Comment by aderusha** [comment]
What's surprising to me is that it failed in the way you experienced.  Typically, a Nextion firmware failure (power loss or whatever) means the firmware update freezes and requires a power cycle of the LCD.  Afterwards, it will usually display an error and wait for you to try another firmware upload.  This is the specific reason for the transistor and resistor on the PCB - they act as a low-side switch to power cycle the Nextion for reasons of being able to remotely recover from a Nextion firmware update failure.

Sounds like the Nextion recovery process isn't as bulletproof as they claimed and I thought.  SD card is always the final approach, I've never had that fail on me so your response to the situation is spot-on.

Thanks for writing this up @ielbury !

**Reply by hockey6611** [reply]
Hi Luma,
I was updating my 2nd HASP (pre-assembled unit) today and got an error similar to what you describe. The first HASP update went smoothly!

I updated HASP firmware to 1.0.0, which went fine. I then went to update nextion display via the WebUI/URL, and it was moving along but at ~20% the screen flashed and now there is a white display with red lettering, "Data ERROR!", and the HASP does not seem to be connected to the network, as the WebUI doesn't connect. I did not have power loss during the update. The device is currently not in-wall, so I can easily power cycle, which does not have any effect.

Will the recovery solution by @ielbury be the best option in this case?

Thank you!

Edit: error text, power cycle clarification

**Reply by aderusha** [reply]
If it’s not in the wall and you have access to the SD card, then sure!  Otherwise, a power cycle should bring back the network functionality, and the LCD will probably just show a white error screen.  Once the network is back online, you can try again.

But if you can do SD, it’s by far the fastest solution and is also pretty reliable.

**Reply by hockey6611** [reply]
Thank you for the reply!

I was able to update the nextion firmware finally, after about 10 attempts.

(Apologies to @ielbury that this issue is now appears unrelated to the original post. Please let me know if I should start a new issue.)
Now that the screen is working, I think I can see the issue I was having updating the display.

The HASP appear to be disconnecting from the network every 5-10 minutes. After it disconnects it is unresponsive for a couple minutes, then the "Wifi Connecting..." screen is shown that it is connecting to the network, then after a couple minutes the new device screen is displayed and the HASwitchPlate wifi network is broadcasted (interestingly, I cannot connect to the setup wifi network at this point).

When I turn off and then back on quickly, the "Wifi Connecting..." screen show and it never connects. When I leave if off for a >5 minutes, after powering on, it immediately connects to the network and mqtt. It operates normally until ~5-10 minutes when it disconnects again.

I have tried rebooting and updating esp from URL, with no changes.

I have lastly tried updating esp from file in the web UI using the bin file. However, now it appears to be bricked. The screen is stuck on "HASP Initializing...".

**Reply by aderusha** [reply]
What are you using for WiFi in your environment?  See [this thread](https://github.com/HASwitchPlate/HASPone/discussions/12) where a few people were having similar problems w/ Ubiquiti APs.

**Reply by hockey6611** [reply]
Thank you! I didn't see this yet. It does sound like my problem (Unifi AP ✔, dhcp reservations ✔, pfsense router though). Interestingly, I have not had any issues with my first HASP device running V1.0 firmware. I will dig more into that thread and reply there with any success or further issue.
